src/ray/raylet/agent_manager.cc

```diff
@@ -16,25 +16,44 @@
 
 #include <thread>
 
+#include "nlohmann/json.hpp"
 #include "ray/common/ray_config.h"
 #include "ray/util/logging.h"
 #include "ray/util/process.h"
 
+using json = nlohmann::json;
+
 namespace ray {
 namespace raylet {
 
 void AgentManager::HandleRegisterAgent(const rpc::RegisterAgentRequest &request,
                                        rpc::RegisterAgentReply *reply,
                                        rpc::SendReplyCallback send_reply_callback) {
-  agent_ip_address_ = request.agent_ip_address();
-  agent_port_ = request.agent_port();
-  agent_pid_ = request.agent_pid();
-  runtime_env_agent_client_ =
-      runtime_env_agent_client_factory_(agent_ip_address_, agent_port_);
-  RAY_LOG(INFO) << "HandleRegisterAgent, ip: " << agent_ip_address_
-                << ", port: " << agent_port_ << ", pid: " << agent_pid_;
-  reply->set_status(rpc::AGENT_RPC_STATUS_OK);
-  send_reply_callback(ray::Status::OK(), nullptr, nullptr);
+  RAY_LOG(INFO) << "HandleRegisterAgent, ip: " << request.agent_ip_address()
+                << ", port: " << request.agent_port() << ", pid: " << request.agent_pid();
+  json address;
+  address.push_back(request.agent_ip_address());
+  address.push_back(request.agent_port());
+  const std::string address_str = address.dump();
+  put_agent_address_(
```

Comment:
can we just put the protobuf in there? no arbitrary json construction and parsing is involved.

Reply:
Good suggestion. I will change the json to protobuf.